modules/gltools/src/state-tracker/unified-parameter-api.js

```diff
@@ -92,35 +71,51 @@ export function getParameters(gl, parameters) {
 
   const state = {};
   for (const key of parameterKeys) {
-    state[key] = getParameter(gl, key);
+    const getter = GL_PARAMETER_GETTERS[key];
+    state[key] = getter ? getter(gl, Number(key)) : gl.getParameter(Number(key));
   }
   return state;
 }
 
-export function getDefaultParameters(gl) {
-  // TODO - Query GL.VIEWPORT and GL.SCISSOR_BOX since these are dynamic
-  return Object.assign({}, GL_PARAMETER_DEFAULTS, {
-    // TODO: For viewport and scissor default values are set at the time of
-    // context creation based on canvas size, we can query them here but it will
-    // not match with what we have in GL_PARAMETER_DEFAULTS table, we should revisit.
-    // [GL.VIEWPORT]: gl.constructor.prototype.getParameter.call(gl, GL.VIEWPORT),
-    // [GL.SCISSOR_BOX]: gl.constructor.prototype.getParameter.call(gl, GL.SCISSOR_BOX)
-  });
-}
-
 // Reset all parameters to a pure context state
 export function resetParameters(gl) {
-  setParameters(gl, getDefaultParameters(gl));
+  setParameters(gl, GL_PARAMETER_DEFAULTS);
```

Comment:
Keep the comment that explains that the defaults will not be correct for certain values, e.g. those that are initialized based on canvas sizes?